src/requestHandlers/panelSet.ts

```diff
@@ -234,68 +249,36 @@ const _getTreeController = (sequelize: Sequelize) => async(id: number) => {
-const dumbDumb = async (request: Request, res: Response) => {
-    const response = await _dumbDumbController(sequelize)();
-    return sanitizeResponse(response, res, '');
-};
 
-const _dumbDumbController = (sequelize: Sequelize) => async() => {
-    try {
+/**
+ * Throws an error if the next_panel_set is invalid
+ * @param next_panel_set_id panel set id of the next panel set the hook is attached to
+ */
+const validateHookConnection = (sequelize : Sequelize) => async(next_panel_set_id : number, validateTrunks : boolean) =>{
 
-        // create a user
-        const user = await UserService.createUser(sequelize)({
-            password:     'Password!',
-            email:        'email@yaoo.com',
-            display_name: 'display'
-        }) as IUser;
-
-        // create panel sets
-        const panel_set_count = 10;
-        const panel_sets = [];
-        for (let i = 0; i < panel_set_count; i++) {
-            panel_sets.push(await PanelSetService.createPanelSet(sequelize)({ author_id: user.id }) as IPanelSet);
-        }
-        const panel_data = [1, 2, 3, 4, 4, 4];
-
-        // create 1 panel for each panel set
-        for (let i = 0; i < panel_data.length; i++) {
-            await PanelService.createPanel(sequelize)({
-                image:        '',
-                index:        0,
-                panel_set_id: panel_data[i],
-            }) as IPanel;
-        }
+    // get panel set to connect to
+    const panelSet = await sequelize.models.panel_set.findByPk(next_panel_set_id, { include: sequelize.models.hook }) as IPanelSet;
 
-        const hookConnection = [
-            { panel_id: 1, next_panel_set_id: 2 },
-            { panel_id: 1, next_panel_set_id: 3 },
-            { panel_id: 2, next_panel_set_id: 4 },
-            { panel_id: 4, next_panel_set_id: 5 },
-            { panel_id: 5, next_panel_set_id: 6 },
-            { panel_id: 6, next_panel_set_id: 7 }
-        ];
-
-        // add hooks
-        for (const hook of hookConnection) {
-            await HookService.createHook(sequelize)({
-                position: {
-                    conditions: {},
-                    path:       '',
-                    value:      ''
-                },
-                current_panel_id:  hook.panel_id,
-                next_panel_set_id: hook.next_panel_set_id
-            });
-        }
+    // if no panel set is found
+    if (panelSet == null) throw new Error('no panel_set exists for given panel_set_id');
 
-        return { success: true };
-    }
-    catch (err) {
-        return err;
+    // if panel set is hooked up
+    if (panelSet.hook) throw new Error('Panel set already has a connection, it cannot be connected to anything else.');
+
+    if (validateTrunks === true) {
+
+        // get the trunks
+        const trunks = await _getAllTrunkSetsController(sequelize) as IPanelSet[];
+
+        // check if the trunk has it
+        if (trunks.map(trunk => trunk.id).some(id => id === panelSet.id)) {
```

Comment:
```suggestion
        if (trunks.some(trunk => trunk.id === panelSet.id)) {
```
I would verify that I wrote the syntax correctly.